kong/runloop/handler.lua

```diff
@@ -78,13 +78,26 @@ do
     return false
   end
 
+
+  local function load_service_from_db(service_pk)
+    return kong.db.services:select(service_pk)
+  end
+
   local function get_service_for_route(db, route)
     local service_pk = route.service
     if not service_pk then
       return nil
     end
 
-    local service, err = db.services:select(service_pk)
+    local service, err
+    if kong.cache then -- the kong.cache module is sometimes missing during tests
+      local cache_key = kong.db.services:cache_key(service_pk.id)
```

Comment:
`db` is already given as an argument to this function, better skip the `kong.` lookup cost